src/test/java/org/scijava/log/MultiThreadingTest.java

```diff
@@ -0,0 +1,78 @@
+package org.scijava.log;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Check if calling {@link AbstractLogService#channel(String)} always only returns one Logger object per name,
+ * even if multiple Threads try to create the same channels in parallel.
+ *
+ * @author Matthias Arzt
+ */
+public class MultiThreadingTest {
+
+	private LogService logService;
+
+	@Before
+	public void setUp() {
+		logService = new StderrLogService();
+	}
+
+	@Test
+	public void run() throws InterruptedException {
+		List<List<Logger>> results = runParallelCollectResults(10, this::getLoggers);
+		boolean passed = allEqual(results, MultiThreadingTest::referencesInListEqual);
+		assertTrue(passed);
```

Comment:
Could collapse this into an `assertAllEqual` static method in `org.scijava.test.TestUtils`. Of course, there is value in having `allEqual` which returns `boolean` as well. How about if that lives in `org.scijava.util` somewhere? Right now we have `ArrayUtils` and `ListUtils`. However, we are moving more toward the ImgLib2-style (and core Java style) of plurals for utility classes. Any suggestions? Happy to move/reorganize/deprecate existing stuff into better structure along these lines.